### PR TITLE
fix(SAT): Adding full negative integer support for log ints

### LINKED ISSUE #1060: Add full negative support and overflow handling to SAT integer rules
At the moment the SAT integer rules don't all work correctly for negative integers and do not have checks for integer over/underflow. Adding support for these should be fairly simple but does require fully understanding how the `SATInt` expression and its associated representation work as well as how the individual rules are written.

Integration tests should also be added for all new edge cases.

### PR DESCRIPTION
**Pull request by hecent**
## Description
Currently not all of the log integer methods handle negatives  correctly, this PR should fix this. It also introduces dedicated selection functions: `tseytin_mux` for conditional booleans - ` cond ? b : a` and `tseytin_select_array` for performing conditionals with array of booleans (for integer representations).

Closes #1060.

### CODE REVIEW

File: crates/conjure-cp-rules/src/sat/log_int_ops.rs [hunk 1]
```diff
@@ -631,12 +621,33 @@ fn tseytin_binary_min_max(
...
-    }
+    tseytin_select_array(mask, x, y, clauses, symbols)
+}
+
+// Selects between two boolean vectors depending on a condition (both vectors must be the same length)
+/// cond ? b : a
+///
+/// cond = 1 => b
+/// cond = 0 => a
+fn tseytin_select_array(
```
Comment (S-andres0694) [inline crates/conjure-cp-rules/src/sat/log_int_ops.rs]: You say that both of the vectors must be the same length. How are you guaranteeing this is the case? In the caller?

Comment (S-andres0694) [reply crates/conjure-cp-rules/src/sat/log_int_ops.rs]: Would you say that there are enough of them for that to become its own PR? Otherwise, maybe we can work together on handling that validation.

Comment (Shikhar-Srivastava-16) [reply crates/conjure-cp-rules/src/sat/log_int_ops.rs]: I would say push a small hotfix right now (one construct should be ok), and then fix it later. Tech debt is fine but lets not kick the can down the line when it comes to functionality.

**Review by ozgurakgun — CHANGES_REQUESTED** [review]
minor hygiene issue, just run `make fix` please
